00-basics/002-javascript/freeCodeCamp_03_19.js

```diff
@@ -0,0 +1,240 @@
+let someObj = {
+    propName: "John"
+};
+
+function propPrefix(str) {
+    let s = "prop";
+    return s + str;
+}
+let someProp = propPrefix("Name");
+console.log(someObj[someProp]);
+
+let testObj = {
+    12: "Namath",
+    16: "Montana",
+    19: "Unitas"
+};
+
+let playerNumber = 16;
+let player = testObj[playerNumber];
+console.log(player);
+
+let myDog = {
+    "name": "Coder",
+    "legs": 4,
+    "tails": 1,
+    "friends": ["freeCodeCamp Campers"]
+};
+myDog.name = "Happy Coder";
+myDog.bark = "woof";
+delete myDog.bark;
+console.log(myDog.name);
+console.log(myDog);
+console.log(myDog);
+
+let myLittleDog = {
+    "name": "Coder",
+    "legs": 4,
+    "tails": 1,
+    "friends": ["freeCodeCamp Campers"]
+};
+
+myDog.bark = "woof";
+delete myDog.bark;
```

Comment:
node that `delete` is one of the most expensive operations in JS (performance-wise), `myDog.bark = undefined` is significantly faster. However, that leaves a property with value `undefined` on the object, which may be undesireable under certain circumstances.